src/routes/solid-router/reference/data-apis/cache.mdx

```diff
@@ -2,16 +2,12 @@
 title: cache
 ---
 
-Cache is used to prevent duplicate fetching and to trigger handle refetching.
-It takes a function and returns the same function.
+## `cache`
 
-```jsx
-const getUser = cache((id) => {
-  return (await fetch(`/api/users${id}`)).json()
-}, "users") // used as cache key + serialized arguments
-```
-
-It is expected that the arguments to the cache function are serializable.
+`cache` is a [higher-order function](https://en.wikipedia.org/wiki/Higher-order_function) that creates a function with the same signature as the function you passed to it.
+When you call the created function with a given set of arguments for the first time, your original function is run and its return value is both stored in a cache and returned to the caller of the created function.
+When you call the created function with the same arguments again before the cache becomes stale, your original function is not re-run—you instead immediately receive the previous return value stored in the cache.
```

Comment:
```suggestion
When the created function is called with the same arguments before the cache becomes stale, the original function is not re-run — instead, it will receive the previous return value stored in the cache.
```